core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/generators/SourceZipWriter.java

```diff
@@ -70,19 +68,19 @@ public File createZip(List<Artifact> artifacts) {
             try (FileOutputStream output = new FileOutputStream(sourceZipPath.toFile());
                     ZipOutputStream zipOutput = new ZipOutputStream(output)) {
                 zipOutput.setLevel(Deflater.BEST_COMPRESSION);
-                for (Artifact artifact : artifacts) {
-                    if (notAllowed.passed(artifact)) {
-                        if (unknownMatchStateFilter.passed(artifact)) {
-                            addContentToZip(artifact, zipOutput);
-                        } else {
-                            if (!artifact.isProprietary()) {
-                                this.reporter.add(artifact,
-                                        MessageType.MATCHSTATE_UNKNOWN,
-                                        "Artifact is not added to sources.zip as MatchState of artifact is declared as unknown.");
+                artifacts.stream()
+                        .filter(artifact -> notAllowed.passed(artifact))
+                        .forEach(artifact -> {
+                            if (unknownMatchStateFilter.passed(artifact)) {
+                                addContentToZip(artifact, zipOutput);
```

Comment:
`addContentToZip` is an IO operation and should throw `IOException`, I am confused why this isn't a problem here

Reply:
I don't understand your question. This is still in a try catch block where an `IOException` is caught.